File: C++/selectionSort.cpp
/*
Program to sort an array using selection sort
Author: Suryansh Maheshwari
Date Modified: 3 October 2021
*/


#include <bits/stdc++.h>
using namespace std;


void selectionSort(int arr[], int n)
{
    int i, j, minIdx;

    for (i = 0; i < n - 1; i++)
    {
        // find the minimum element in unsorted array
        minIdx = i;
        for (j = i + 1; j < n; j++)
            if (arr[j] < arr[minIdx])
                minIdx = j;

        if (minIdx != i)
            swap(arr[minIdx], arr[i]);
    }
}

/* function to print an array */
void printArray(int arr[], int size)
{
    int i;
    for (i = 0; i < size; i++)
        cout << arr[i] << " ";
    cout << endl;
}


int main()
{
    int arr[] = {64, 25, 19, 4, 13};
    int n = sizeof(arr) / sizeof(arr[0]);
    selectionSort(arr, n);

    cout << "Sorted array: \n";
    printArray(arr, n);

    return 0;
}

File: python/Selection_Sort.py
# SELECTION SORT
# AUTHOR: SAURABH GUPTA
# DATE MODIFIED: 6 OCT 2021

def selection_sort(arr):
    for i in range(0, len(arr) - 1):
        smallest = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[smallest]:
                smallest = j
        arr[i], arr[smallest] = arr[smallest], arr[i]
    return arr

lst = []
size = int(input("Enter size of the list: "))
elements=[]
elements = (input("Enter the elements: ")).split(" ")
for i in range(0,len(elements)):
    lst.append(int(elements[i]))
 
print(selection_sort(lst))
